### PR TITLE
Run containers as non-root & reduce the size of pcw image

### PR DESCRIPTION
**Pull request by ricardobranco777**
This PR:
- Makes it possible to run the containers as non-root
- Adds `--no-cache-dir` to `pip install` to reduce the size by 53M

Fixes: https://github.com/SUSE/pcw/issues/206

Related MR:
https://gitlab.suse.de/qac/publiccloud-qa-suse-de/-/merge_requests/70

### CODE REVIEW

**Comment by pdostal** [comment]
Is `--no-cache-dir` necessary when we have `&& rm -rf /var/cache` at the end of that command?

File: Dockerfile [hunk 1]
```diff
@@ -28,5 +28,8 @@ VOLUME /pcw/db
 
 EXPOSE 8000/tcp
 
+RUN useradd --no-create-home --uid 1777 --user-group --shell /bin/false pcw && chown -R pcw:pcw /pcw
+USER pcw
```
Comment (pdostal) [inline Dockerfile:32]: This is unrelated to current pull request. I'm all for it but I think this should be separate.

**Review by pdostal — APPROVED** [review]

File: Dockerfile_dev [hunk 2]
```diff
@@ -10,7 +10,7 @@ COPY requirements.txt requirements_test.txt requirements_k8s.txt /tmp/
 # * Install system requirements
 # * Install pip requirements
 # * Empty system cache to conserve some space
-RUN zypper -n in python310-devel gcc libffi-devel aws-cli && pip3.10 install -r /tmp/requirements_test.txt && rm -rf /var/cache
+RUN zypper -n in python310-devel gcc libffi-devel aws-cli && pip install --no-cache-dir -r /tmp/requirements_test.txt && rm -rf /var/cache
```
[the remark this answers is not in the document]
Comment (ricardobranco777) [reply Dockerfile_dev:14]: > what about _dev container ? I think it should follow same non-root approach so we will catch any user-related problems earlier but I am not sure that it is right thing to use `pcw` user which certainly not exists anywhere . maybe for _dev container we need to build it with user who init `podman build ` ? I think it is possible to propagate via Makefile ... WDYT ?

The related MR (that was already merged by @pdostal takes care of this). I tried this PR & MR it and it works.

Will add it to _dev as well.

File: Dockerfile [hunk 3]
```diff
@@ -10,7 +10,7 @@ COPY requirements.txt /pcw/
 # * Install system requirements
 # * Install pip requirements
 # * Empty system cache to conserve some space
-RUN zypper -n in python310-devel gcc libffi-devel aws-cli && pip3.10 install -r /pcw/requirements.txt && rm -rf /var/cache
+RUN zypper -n in python310-devel gcc libffi-devel aws-cli && pip install --no-cache-dir -r /pcw/requirements.txt && rm -rf /var/cache
```
Comment (asmorodskyi) [inline Dockerfile:13]: if you want to introduce user I think you also need to solve this 
```
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
```

Comment (asmorodskyi) [reply Dockerfile:13]: but instead of virtualenv I would simply run pip AFTER switching to pcw user , don't think it make sense to create virtualenv inside container

Comment (asmorodskyi) [reply Dockerfile:13]: ok forget it
